src/main/java/org/sonar/plugins/stash/issue/collector/StashCollector.java

```diff
@@ -99,150 +102,180 @@ public static StashPullRequest extractPullRequest(String project, String reposit
+      // Let's call this for loop "objhunk_loop"
+      for (Object objHunk : jsonHunks.toArray()) {
 
-              for (Object objLineComment : jsonLineComments.toArray()) {
-                JSONObject jsonLineComment = (JSONObject) objLineComment;
+        JSONObject jsonHunk    = (JSONObject)objHunk;
+        JSONArray jsonSegments = (JSONArray)jsonHunk.get("segments");
 
-                long lineCommentId = (long) jsonLineComment.get("id");
-                String lineCommentMessage = (String) jsonLineComment.get("text");
-                long lineCommentVersion = (long) jsonLineComment.get(VERSION);
+        if (jsonSegments == null) { continue; } // Let's process the next item in "objhunk_loop"
 
-                JSONObject objAuthor = (JSONObject) jsonLineComment.get(AUTHOR);
-                if (objAuthor != null) {
+        // Let's call this for loop "objsegm_loop"
+        for (Object objSegment : jsonSegments.toArray()) {
 
-                  StashUser author = extractUser(objAuthor);
+          JSONObject jsonSegment = (JSONObject)objSegment;
+          // type of the diff in diff view
+          // We filter REMOVED type, like useless for SQ analysis
+          String type = (String)jsonSegment.get("type");
 
-                  StashComment comment = new StashComment(lineCommentId, lineCommentMessage, path, (long) 0, author, lineCommentVersion);
-                  initialDiff.addComment(comment);
-                }
-              }
+          // Let's process the next item in "objsegm_loop"
+          if (StringUtils.equals(type, StashPlugin.REMOVED_ISSUE_TYPE)) { continue; }
 
-              result.add(initialDiff);
+          JSONArray jsonLines = (JSONArray)jsonSegment.get("lines");
+
+          if (jsonLines == null) { continue; } // Let's process the next item in "objsegm_loop"
+
+          // Let's call this for loop "objline_loop"
+          for (Object objLine : jsonLines.toArray()) {
+
+            JSONObject jsonLine = (JSONObject)objLine;
+            // destination line in diff view
+            long source = (long)jsonLine.get("source");
+            long destination = (long)jsonLine.get("destination");
+
+            StashDiff diff = new StashDiff(type, path, source, destination);
+            // Add comment attached to the current line
+            JSONArray jsonCommentIds = (JSONArray)jsonLine.get("commentIds");
+
+            if (jsonCommentIds == null) {
+              result.add(diff);
+            } else {
+              // To keep this method depth under control (squid:S134), we outsourced the comments extraction
+              result.add(extractCommentsForDiff(diff, jsonDiff, jsonCommentIds));
             }
           }
         }
       }
-    }
 
+      // Extract File Comments: this kind of comment will be attached to line 0
+      JSONArray jsonLineComments = (JSONArray)jsonDiff.get("fileComments");
+
+      if (jsonLineComments == null) { continue; } // Let's process the next item in "objdiff_loop"

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)